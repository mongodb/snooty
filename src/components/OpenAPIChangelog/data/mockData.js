export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
