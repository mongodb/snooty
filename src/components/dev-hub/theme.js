/**
 * @type {Object.<string, string>}
 * @property {string} fontsize returns px value
 */
const fontSize = {
  xsmall: '10px',
  tiny: '12px',
  small: '14px',
  default: '16px',
  h1: '48px',
  h2: '30px',
  h3: '24px',
  h4: '20px',
  tabNav: '10px',
};
/**
 * @type {Object}
 * @property {string} size returns px value
 * @property {function(string): number} stripUnit removes px unit
 */
const size = {
  tiny: '5px',
  small: '10px',
  default: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '64px',
  xxlarge: '128px',
  maxWidth: '1400px',
};
const colorMap = {
  black: '#1B272D',
  darkGreen: '#13AA52',
  devBlack: '#1a282e',
  devWhite: '#e3ecea',
  greyDarkOne: '#5B6D75',
  greyDarkTwo: '#3A5058',
  greyDarkThree: '#2A3D46',
  greyLightOne: '#E3ECEA',
  greyLightTwo: '#B4C4C1',
  greyLightThree: '#85979A',
  lightGreen: '#0AD05B',
  orange: '#ED7E22',
  magenta: '#CD509B',
  salmon: '#E55F55',
  teal: '#2F9FC5',
  violet: '#6E60F9',
  white: '#FFFFFF',
  yellow: '#FDDC49',
};

/**
 * store common responsive sizes
 * @type {Object.<string, string>}
 */
const screenSize = {
  upToXSmall: 'only screen and (max-width: 374px)',
  xSmallAndUp: 'not all and (max-width: 374px)',
  upToSmall: 'only screen and (max-width: 420px)',
  smallAndUp: 'not all and (max-width: 420px)',
  upToMedium: 'only screen and (max-width: 767px)',
  mediumAndUp: 'not all and (max-width: 767px)',
  upToLarge: 'only screen and (max-width: 1023px)',
  largeAndUp: 'not all and (max-width: 1023px)',
};

export { colorMap, fontSize, screenSize, size };
