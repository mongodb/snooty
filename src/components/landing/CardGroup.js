import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';

const getColumnValue = (props) => props.columns || React.Children.count(props.children);

const carouselStyling = ({ children, theme }) => css`
  grid-gap: calc(${theme.size.medium} * 0.75);
  grid-template-columns:
    calc(${theme.size.medium} / 2) repeat(${React.Children.count(children)}, calc(75% - calc(2 * ${theme.size.medium})))
    calc(${theme.size.medium} / 2);
  grid-template-rows: minmax(150px, 1fr);
  margin: ${theme.size.medium} 0;
  overflow-x: scroll;
  padding-bottom: calc(${theme.size.medium} / 2);
  scroll-snap-type: x proximity;

  &:before,
  &:after {
    content: '';
  }
`;

const StyledGrid = styled('div')`
  align-items: stretch;
  display: grid;
  grid-column: 1/-1 !important;
  grid-column-gap: ${({ theme }) => theme.size.medium};
  grid-row-gap: ${({ theme }) => theme.size.medium};
  grid-template-columns: ${(props) => `repeat(${getColumnValue(props)}, 1fr)`};
  margin: ${({ theme }) => theme.size.large} 0;

  @media ${({ theme }) => theme.screenSize.upToLarge} {
    grid-template-columns: repeat(2, 1fr);
  }

  @media ${({ theme }) => theme.screenSize.upToMedium} {
    ${({ isCarousel, ...props }) => (isCarousel ? carouselStyling(props) : 'grid-template-columns: repeat(1, 1fr);')}
  }
`;

const CardGroup = ({
  nodeData: {
    children,
    options: { columns, layout, style },
  },
  ...rest
}) => {
  const isCompact = style === 'compact';
  const isExtraCompact = style === 'extra-compact';
  const isCarousel = layout === 'carousel';
  return (
    <StyledGrid columns={columns} noMargin={true} isCarousel={isCarousel}>
      {children.map((child, i) => (
        <ComponentFactory {...rest} key={i} nodeData={child} isCompact={isCompact} isExtraCompact={isExtraCompact} />
      ))}
    </StyledGrid>
  );
};

CardGroup.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object),
    options: PropTypes.shape({
      columns: PropTypes.number,
    }),
  }).isRequired,
};

export default CardGroup;
