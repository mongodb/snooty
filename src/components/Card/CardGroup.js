import React from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';
import { theme } from '../../theme/docsTheme';

const getMarginStyles = (isForDriver, isLanding, columns, sideMarginValue) => {
  if (typeof sideMarginValue !== 'number') {
    console.warn('sideMarginValue only accepts a number');
  }
  if (isForDriver) return `0 ${sideMarginValue}px ${theme.size.xlarge}`;
  else if (isLanding && columns !== 3) return `${theme.size.large} ${sideMarginValue}px`;
  else if (isLanding && columns === 3) return `${theme.size.medium} ${sideMarginValue}px ${theme.size.xlarge}`;
  else return `${theme.size.large} 0`;
};

const getColumnValue = (props) => props.columns || React.Children.count(props.children);

// Carousel styling refers to the horizontal scrolling display of the Cards;
// This is not a true carousel as it is not 'circular,' but is rather 1 row
// with n columns displaying only a subset on screen
const carouselStyling = ({ children, theme }) => css`
  grid-gap: calc(${theme.size.medium} * 0.75);
  grid-template-columns:
    calc(${theme.size.medium} / 2) repeat(${React.Children.count(children)}, calc(75% - calc(2 * ${theme.size.medium})))
    calc(${theme.size.medium} / 2);
  grid-template-rows: minmax(150px, 1fr);
  margin: ${theme.size.medium} 0;
  overflow-x: scroll;
  padding-bottom: calc(${theme.size.medium} / 2);
  scroll-snap-type: x proximity;

  &:before,
  &:after {
    content: '';
  }
`;

const verticalStyling = css`
  grid-column-gap: ${theme.size.default};
  grid-template-columns: 'auto';
  grid-row-gap: '${theme.size.default}';
`;

// StyledGrid behavior on medium and small screens is determined by 'isCarousel'
const StyledGrid = styled('div')`
  align-items: stretch;
  display: grid;
  grid-column: 1 / -1 !important;
  grid-column-gap: ${theme.size.medium};
  grid-row-gap: ${theme.size.medium};
  grid-template-columns: ${(props) => `repeat(${getColumnValue(props)}, 1fr)`};

  margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(isForDrivers, isLanding, columns, 0)};

  @media ${theme.screenSize.upToMedium} {
    margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(isForDrivers, isLanding, columns, 42)};
  }

  @media ${theme.screenSize.upToSmall} {
    margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(isForDrivers, isLanding, columns, 24)};
  }

  ${
    '' /*want first 4 cards on landing page to stay as one column, and not affect group of 3 cards (so hacky i'm sorry) */
  }
  @media ${theme.screenSize.upToXLarge} {
    ${({ isLanding, columns }) => {
      if (isLanding) return `grid-template-columns: repeat(${columns}, 1fr);`;
      else return `grid-template-columns: repeat(2, 1fr);`;
    }}

  @media ${theme.screenSize.upToMedium} {
    ${({ isCarousel, ...props }) =>
      isCarousel
        ? carouselStyling(props)
        : `grid-template-columns: repeat(1, 1fr); grid-column-gap: ${theme.size.default};`}
  }

  ${'' /* If not isCarousel, stack Cards vertically in 1 column on small screens */}
  @media ${theme.screenSize.upToSmall} {
    ${({ isCarousel }) => !isCarousel && verticalStyling}
  }
`;

const CardGroup = ({
  className,
  nodeData: {
    children,
    options: { columns, layout, style, type },
  },
  page,
  ...rest
}) => {
  const isCompact = style === 'compact';
  const isExtraCompact = style === 'extra-compact';
  const isCarousel = layout === 'carousel';
  const isForDrivers = type === 'drivers';
  const isLanding = page?.options?.template === 'landing';

  return (
    <StyledGrid
      className={[
        'card-group',
        isCompact ? 'compact' : '',
        isExtraCompact ? 'extra-compact' : '',
        isForDrivers ? 'drivers' : '',
        className,
      ].join(' ')}
      columns={columns}
      noMargin={true}
      isCarousel={isCarousel}
      isForDrivers={isForDrivers}
      isLanding={isLanding}
    >
      {children.map((child, i) => (
        <ComponentFactory
          {...rest}
          key={i}
          nodeData={child}
          isCompact={isCompact}
          isExtraCompact={isExtraCompact}
          isForDrivers={isForDrivers}
          page={page}
        />
      ))}
    </StyledGrid>
  );
};

CardGroup.propTypes = {
  className: PropTypes.string,
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object),
    options: PropTypes.shape({
      columns: PropTypes.number,
    }),
  }).isRequired,
  page: PropTypes.object,
};

export default CardGroup;
