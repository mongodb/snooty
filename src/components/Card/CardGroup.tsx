import React from 'react';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import ComponentFactory from '../ComponentFactory';
import { theme } from '../../theme/docsTheme';
import type { CardGroupNode } from '../../types/ast';
import { Page } from '../../context/page-context';

export interface CardGroupProps {
  className?: string;
  nodeData: CardGroupNode;
  page?: Page;
}

interface StyledGridProps {
  isForDrivers: boolean;
  isLanding: boolean;
  columns: number;
  isCarousel?: boolean;
  className?: string;
  noMargin?: boolean;
}

const getMarginStyles = (
  isForDriver: boolean,
  isLanding: boolean,
  columns: number,
  sideMarginValue: number
): string => {
  if (typeof sideMarginValue !== 'number') {
    console.warn('sideMarginValue only accepts a number');
  }
  if (isForDriver) return `0 ${sideMarginValue}px ${theme.size.xlarge}`;
  else if (isLanding && columns !== 3) return `${theme.size.large} ${sideMarginValue}px`;
  else if (isLanding && columns === 3) return `${theme.size.medium} ${sideMarginValue}px ${theme.size.xlarge}`;
  else return `${theme.size.large} 0`;
};

const getColumnValue = (props: { columns?: number; children?: React.ReactNode }) =>
  props.columns || React.Children.count(props.children);

// Carousel styling refers to the horizontal scrolling display of the Cards;
// This is not a true carousel as it is not 'circular,' but is rather 1 row
// with n columns displaying only a subset on screen
const carouselStyling = ({ children }: { children?: React.ReactNode }) => css`
  grid-gap: calc(${theme.size.medium} * 0.75);
  grid-template-columns:
    calc(${theme.size.medium} / 2) repeat(${React.Children.count(children)}, calc(75% - calc(2 * ${theme.size.medium})))
    calc(${theme.size.medium} / 2);
  grid-template-rows: minmax(150px, 1fr);
  margin: ${theme.size.medium} 0;
  overflow-x: scroll;
  padding-bottom: calc(${theme.size.medium} / 2);
  scroll-snap-type: x proximity;

  &:before,
  &:after {
    content: '';
  }
`;

const verticalStyling = css`
  grid-column-gap: ${theme.size.default};
  grid-template-columns: 'auto';
  grid-row-gap: '${theme.size.default}';
`;

// StyledGrid behavior on medium and small screens is determined by 'isCarousel'
const StyledGrid = styled('div')<StyledGridProps>`
  align-items: stretch;
  display: grid;
  grid-column: 1 / -1 !important;
  grid-column-gap: ${theme.size.medium};
  grid-row-gap: ${theme.size.medium};
  grid-template-columns: ${(props) => `repeat(${getColumnValue(props)}, 1fr)`};

  margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(Boolean(isForDrivers), isLanding, columns, 0)};

  @media ${theme.screenSize.upToMedium} {
    margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(isForDrivers, isLanding, columns, 42)};
  }

  @media ${theme.screenSize.upToSmall} {
    margin: ${({ isForDrivers, isLanding, columns }) => getMarginStyles(isForDrivers, isLanding, columns, 24)};
  }

  ${
    '' /*want first 4 cards on landing page to stay as one column, and not affect group of 3 cards (so hacky i'm sorry) */
  }
  @media ${theme.screenSize.upToXLarge} {
    ${({ isLanding, columns }) => {
      if (isLanding) return `grid-template-columns: repeat(${columns}, 1fr);`;
      else return `grid-template-columns: repeat(2, 1fr);`;
    }}

  @media ${theme.screenSize.upToMedium} {
    ${({ isCarousel, ...props }) =>
      isCarousel
        ? carouselStyling(props)
        : `grid-template-columns: repeat(1, 1fr); grid-column-gap: ${theme.size.default};`}
  }

  ${'' /* If not isCarousel, stack Cards vertically in 1 column on small screens */}
  @media ${theme.screenSize.upToSmall} {
    ${({ isCarousel }) => !isCarousel && verticalStyling}
  }
`;

const CardGroup = ({
  className,
  nodeData: {
    children = [],
    options: { columns, layout, style, type },
  },
  page,
  ...rest
}: CardGroupProps) => {
  const isCompact = style === 'compact';
  const isExtraCompact = style === 'extra-compact';
  const isCarousel = layout === 'carousel';
  // Keep "type" for backwards compatibility, but it might be good to generalize to "style"
  const isCenterContentStyle = type === 'drivers' || style === 'center-content';
  const isLargeIconStyle = style === 'large-icon';
  const isDriversTemplate = page?.options?.template === 'drivers-index';
  const isLanding = page?.options?.template === 'landing';

  return (
    <StyledGrid
      className={['card-group', className].join(' ')}
      columns={columns}
      noMargin={true}
      isCarousel={isCarousel}
      isForDrivers={isDriversTemplate}
      isLanding={isLanding}
    >
      {children.map((child, i) => (
        <ComponentFactory
          {...rest}
          key={i}
          nodeData={child}
          isCompact={isCompact}
          isExtraCompact={isExtraCompact}
          isCenterContentStyle={isCenterContentStyle}
          isLargeIconStyle={isLargeIconStyle}
          page={page}
        />
      ))}
    </StyledGrid>
  );
};

export default CardGroup;
