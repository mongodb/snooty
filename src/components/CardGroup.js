import React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import ComponentFactory from './ComponentFactory';
import { theme } from '../theme/docsTheme';

const getColumnValue = (props) => props.columns || React.Children.count(props.children);

// Carousel styling refers to the horizontal scrolling display of the Cards;
// This is not a true carousel as it is not 'circular,' but is rather 1 row
// with n columns displaying only a subset on screen
const carouselStyling = ({ children, theme }) => css`
  grid-gap: calc(${theme.size.medium} * 0.75);
  grid-template-columns:
    calc(${theme.size.medium} / 2) repeat(${React.Children.count(children)}, calc(75% - calc(2 * ${theme.size.medium})))
    calc(${theme.size.medium} / 2);
  grid-template-rows: minmax(150px, 1fr);
  margin: ${theme.size.medium} 0;
  overflow-x: scroll;
  padding-bottom: calc(${theme.size.medium} / 2);
  scroll-snap-type: x proximity;

  &:before,
  &:after {
    content: '';
  }
`;

const verticalStyling = css`
  grid-column-gap: ${theme.size.default};
  grid-template-columns: 'auto';
  grid-row-gap: '${theme.size.default}';
`;

// StyledGrid behavior on medium and small screens is determined by 'isCarousel'
const StyledGrid = styled('div')`
  align-items: stretch;
  display: grid;
  grid-column: 1 / -1 !important;
  grid-column-gap: ${theme.size.medium};
  grid-row-gap: ${theme.size.medium};
  grid-template-columns: ${(props) => `repeat(${getColumnValue(props)}, 1fr)`};
  margin: ${theme.size.large} 0;

  @media ${theme.screenSize.upToLarge} {
    grid-template-columns: repeat(2, 1fr);
  }

  @media ${theme.screenSize.upToMedium} {
    ${({ isCarousel, ...props }) =>
      isCarousel
        ? carouselStyling(props)
        : `grid-template-columns: repeat(1, 1fr); grid-column-gap: ${theme.size.default};`}
  }

  ${'' /* If not isCarousel, stack Cards vertically in 1 column on small screens */}
  @media ${theme.screenSize.upToSmall} {
    ${({ isCarousel }) => !isCarousel && verticalStyling}
  }
`;

const CardGroup = ({
  nodeData: {
    children,
    options: { columns, layout, style },
  },
  ...rest
}) => {
  const isCompact = style === 'compact';
  const isExtraCompact = style === 'extra-compact';
  const isCarousel = layout === 'carousel';
  return (
    <StyledGrid className="card-group" columns={columns} noMargin={true} isCarousel={isCarousel}>
      {children.map((child, i) => (
        <ComponentFactory {...rest} key={i} nodeData={child} isCompact={isCompact} isExtraCompact={isExtraCompact} />
      ))}
    </StyledGrid>
  );
};

CardGroup.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object),
    options: PropTypes.shape({
      columns: PropTypes.number,
    }),
  }).isRequired,
};

export default CardGroup;
