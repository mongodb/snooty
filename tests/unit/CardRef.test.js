import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'emotion-theming';
import CardGroup from '../../src/components/CardGroup';
import { theme } from '../../src/theme/docsTheme';

// data for this component
import mockData from './data/CardRef.test.json';

// Since there isn't a proper CardRef component, test that card-ref styling
// is applied appropriately (i.e., only when a Card :url: is not specified)
it('card correctly with and without url', () => {
  const tree = mount(
    <ThemeProvider theme={theme}>
      <CardGroup nodeData={mockData} />
    </ThemeProvider>
  ).childAt(0);
  expect(tree).toMatchSnapshot();
});
