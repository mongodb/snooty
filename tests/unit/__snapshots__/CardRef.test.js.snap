// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card correctly with and without url 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: grid;
  grid-column: 1/-1!important;
  grid-column-gap: 24px;
  grid-row-gap: 24px;
  grid-template-columns: repeat(3, 1fr);
  margin: 32px 0;
}

@media only screen and (max-width: 1200px) {
  .emotion-0 {
    grid-template-columns: repeat(2, 1fr);
  }
}

@media only screen and (max-width: 767px) {
  .emotion-0 {
    grid-template-columns: repeat(1, 1fr);
    grid-column-gap: 16px;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-0 {
    grid-column-gap: 16px;
    grid-template-columns: 'auto';
    grid-row-gap: '16px';
  }
}

.emotion-2 {
  position: relative;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: border,box-shadow;
  border: 1px solid #E8EDEB;
  box-shadow: 0 4px 10px -4px rgba(0,30,43,0.3);
  background-color: white;
  color: #1C2D38;
  border-radius: 24px;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 13px;
  line-height: 20px;
  padding: 24px;
  min-height: 68px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 32px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 32px 24px;
}

.emotion-2 p:last-of-type {
  margin-bottom: 0;
}

.emotion-3 {
  width: 24px;
}

@media only screen and (max-width: 480px) {
  .emotion-3 {
    width: 20px;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  margin-left: 8px;
}

@media only screen and (max-width: 480px) {
  .emotion-5 {
    margin-left: 16px;
  }
}

.emotion-5 p {
  line-height: 24px;
}

.emotion-7 {
  margin: unset;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 500;
  font-weight: 600;
  letter-spacing: 0.5px;
  margin: 0 0 8px;
}

.emotion-7 strong,
.emotion-7 b {
  font-weight: 700;
}

.emotion-8 {
  margin: unset;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  margin-bottom: 16px;
}

.emotion-8 strong,
.emotion-8 b {
  font-weight: 700;
}

.emotion-10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
  line-height: 13px;
  outline: none;
  color: #016BF8;
}

.emotion-10:focus {
  outline: none;
}

.emotion-10>code {
  color: #016BF8;
}

.emotion-10::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: -4px;
  left: 0;
  border-radius: 2px;
}

.emotion-10:focus .emotion-10::after {
  background-color: #0498EC;
}

.emotion-10:hover::after {
  background-color: #E8EDEB;
}

.emotion-11 {
  background: #F9FBFA;
  border-radius: 4px;
  border: 1px solid #C1C7C6;
  box-sizing: border-box;
  display: inline-block;
  font-size: 13px!important;
  font-weight: 600;
  margin-bottom: 4px;
  margin-right: 4px;
  padding: 4px;
}

.emotion-11:after {
  content: ' âž”';
}

.emotion-22 {
  position: relative;
  -webkit-transition: 150ms ease-in-out;
  transition: 150ms ease-in-out;
  transition-property: border,box-shadow;
  border: 1px solid #E8EDEB;
  box-shadow: 0 4px 10px -4px rgba(0,30,43,0.3);
  background-color: white;
  color: #1C2D38;
  border-radius: 24px;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 13px;
  line-height: 20px;
  padding: 24px;
  min-height: 68px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 32px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 32px 24px;
}

.emotion-22:focus {
  outline: none;
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC,0 4px 10px -4px rgba(0,30,43,0.3);
}

.emotion-22:hover,
.emotion-22:active {
  border: 1px solid #E8EDEB;
  box-shadow: 0 4px 20px -4px rgba(0,30,43,0.2);
}

.emotion-22:hover:focus,
.emotion-22:active:focus {
  box-shadow: 0 0 0 2px #FFFFFF,0 0 0 4px #0498EC,0 4px 20px -4px rgba(0,30,43,0.2);
}

.emotion-22 p:last-of-type {
  margin-bottom: 0;
}

.emotion-29 {
  margin: unset;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
}

.emotion-29 strong,
.emotion-29 b {
  font-weight: 700;
}

.emotion-29 a {
  line-height: unset;
}

.emotion-30 {
  font-size: 13px;
  line-height: 20px;
  font-family: 'Euclid Circular A',Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  line-height: 13px;
  color: #016BF8;
  font-weight: 400;
  -webkit-text-decoration: none!important;
  text-decoration: none!important;
}

.emotion-30:focus {
  outline: none;
}

.emotion-31 {
  position: relative;
}

.emotion-31::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 2px;
  bottom: -4px;
  left: 0;
  border-radius: 2px;
}

.constant-classname:focus .emotion-31::after {
  background-color: #0498EC;
}

.constant-classname:hover .emotion-31::after {
  background-color: #E8EDEB;
}

<div
    class="card-group  emotion-0 emotion-1"
  >
    <div
      class="emotion-2"
    >
      <img
        class="emotion-3 emotion-4"
        src="/images/icons/realm_sdk.svg"
      />
      <div
        class="emotion-5 emotion-6"
      >
        <p
          class="emotion-7"
        >
          Realm SDKs
        </p>
        <p
          class="emotion-8"
        >
          Use platform-specific libraries to include and work with the Realm database in your application.
        </p>
        <p
          class="emotion-8"
        >
          <a
            class="emotion-10 emotion-11"
            href="/sdk/android/#std-label-android-intro"
          >
            Android
          </a>
          

          <a
            class="emotion-10 emotion-11"
            href="/sdk/ios/#std-label-ios-intro"
          >
            iOS
          </a>
          

          <a
            class="emotion-10 emotion-11"
            href="/sdk/dotnet/#std-label-dotnet-intro"
          >
            .Net
          </a>
          

          <a
            class="emotion-10 emotion-11"
            href="/web/#std-label-web-intro"
          >
            Web
          </a>
          

          <a
            class="emotion-10 emotion-11"
            href="/sdk/react-native/#std-label-react-native-intro"
          >
            React Native
          </a>
          

          <a
            class="emotion-10 emotion-11"
            href="/sdk/node/#std-label-node-intro"
          >
            Node.js
          </a>
        </p>
      </div>
    </div>
    <div
      class="emotion-22"
    >
      <img
        class="emotion-3 emotion-4"
        src="/images/icons/realm.svg"
      />
      <div
        class="emotion-5 emotion-6"
      >
        <p
          class="emotion-7"
        >
          MongoDB Realm
        </p>
        <p
          class="emotion-8"
        >
          Discover how to sync data, define permissions, and connect to other services, including MongoDB Atlas.
        </p>
        <p
          class="emotion-29"
        >
          <a
            class="constant-classname emotion-30"
            href="https://docs.mongodb.com/realm/services/"
            target="_self"
          >
            <span
              class="emotion-31"
            >
              Learn more about MongoDB Realm Application Services
            </span>
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
